### ISSUE TITLE
The role of necessary and sufficient definition methodology

### ISSUE THREAD

**Issue by rogerburks** [insect-morphology/old-aism#10]
In this ontology, we are intentionally departing from the traditional methods of definition, where necessary-and-sufficient criteria are usually used to create an atomized genus-differentia statement. This problem came up in #9 where I was struggling to define "fore leg" in an atomized way, where really we are not trying to atomize definitions. This is a strong intentional departure from the historical philosophy of definitions:

- Normally in definition using traditional methods, we would use methods of differentiation and concept abstraction to atomize the statement. For instance:

"square: a rectangle that is a rhombus."

- This definition is elegant in that it is minimalistic, and eventually is traceable to all statements that could be said about a square. It provides traditional "sufficiency" in that having a square is sufficient to say that you have both some kind of rhombus and some kind of rectangle. However, it does not immediately enumerate all of those statements and therefore is at least slightly more difficult to test than what we are doing. Instead, the above definition uses abstraction to contain within the words "rectangle" and "rhombus" many things that are maintained true about squares. It was therefore up to us to know and fully enumerate all the properties that must be true about squares if a square is simultaneously both rectangle and rhombus.

This is based on defining rectangle and rhombus as abstractions (that is--a word used to embody a more verbose statement) of the following statements:
Rectangle: a quadrilateral that has interior angles which are all right angles.
Rhombus: a quadrilateral that has bounding sides which all have the same length.

- Instead, we are using a different method in the ontology, which breaks out all of those statements and lists them up front, without much use of abstraction, in part because abstraction has been misused or misunderstood historically in descriptive research. Instead we are enumerating statements that are hypotheses, which are subject to rejection if we decide that a statement sometimes does not hold true. Some of these statements would have been traditionally passed over as helpful but emergent properties of the atomized definition. 

What we are doing is something like this, where statements are enumerated one-by-one instead of being abstracted:

A square has 4 sides. (true of class quadrilateral)

A square has right angles. (true of class rectangle)

A square has sides that are all of the same length. (true of class rhombus)

A square has opposing sides that are all parallel. (true of class parallelogram)

- In "square: a rectangle that is a rhombus," all of the above statements are omitted because "square" happens to occupy a point where the definitions of "rectangle" and "rhombus" overlap. Instead of using the traditional method of finding and referencing abstractions, we are enumerating statements that we know about squares. Therefore, we promote the building of verbose definitions with many statements, and we wish to test every statement separately.

- This has the benefit of freeing us from knowing precisely if all insects agree with the usual methods of testing "necessary." Therefore, if mid leg is absent, it does not matter, and a fore leg remains a fore leg. It only matters if we find an insect where the fore leg is not anterior to the mid leg when both happen to be present. This is opposed to traditional [necessity](http://www.sfu.ca/~swartz/conditions1.htm#section9), where one would point out that the mid leg is not necessary to define the fore leg. This method is being used in part because of the amazing and difficult-to-plot diversity of insects, where sclerites and other structures are highly changeable in terms of connection and fusion/separation.

- In other words, this frees us from some possible limitations of induction, such as the [problem of induction](https://en.wikipedia.org/wiki/Problem_of_induction) where we are limited in the power of our definitions by the scarcity of things that we are aware of. Also it deals with [black-swan-theory](https://en.wikipedia.org/wiki/Black_swan_theory). If we previously defined a class "swan" (regardless of species) in part as always being white, and we eventually discover a black swan, we do not reject the class swan, nor do we ignore the black swan. Instead, we would remove the statement about always being white from class swan, since it was rejected.

**Comment by rogerburks** [comment]
To answer the question of "why the heck does Roger care about this?" I have the following:

Let us say that some one made a computer program that automatically labels insect parts and ultimately identifies the species based on 3D scans, using this ontology as part of its foundation. Maybe it is a program meant to identify fossils in visually difficult amber pieces. Let us say that they decided that the logic needed for this is reliant upon necessary criteria, and for some reason the fossil specimen was physically missing a mid leg (maybe it was broken off before preservation).

If the program uses the logic in traditional definitions, it would require that fore leg must be anterior to mid leg. If mid leg was for some reason broken off or otherwise not found by the program, then the statement returns `false` because mid leg is not found, and the fore leg is not recognized as a fore leg.

Instead we are telling the programmer that it needs to work like this: some legs are found and the head is found. If three pairs of legs are found, then the fore leg must arise anterior to the mid leg. 

Of course even in the latter problem, if the mid leg is missing, the fore leg is not recognized using the criterion of placement relative to other legs. However, other criteria can be used which are also not requisite, but which may be helpful in finding the answer. For instance, if muscle attachments are found that can only occur for the fore leg or neighboring parts, then the fore leg is properly identified, even if the mid leg is absent. Or, if special spines or other structures that can only be present on the fore leg are found, then the structure is a fore leg. Because none of these relations is absolutely required to be present but is only required to hold true when tested, the process can more closely resemble human reasoning.
